### PR TITLE
Consolidate Cloud project upgrade topics

### PR DESCRIPTION
**Pull request by hguthrie**
This PR consolidates the [Upgrade Magento version](https://devdocs.magento.com/guides/v2.2/cloud/project/project-upgrade.html) topics for 2.2 and 2.3 into one, generic topic for all versions.

The 2.2 version of the topics contained more sections that assisted with upgrading older projects by converting the `config.php` file and updating the hooks section of the `.magento.app.yaml` file. For this, I broke the topic into two high-level headings.

This may need a technical review, because there are still some one-time-only steps in the topic that may no longer be required for any version.

- [x] Technical review
- [x] Editorial review
- [x] Final link check

### CODE REVIEW

File: guides/v2.2/cloud/project/project-upgrade.md [hunk 1]
```diff
@@ -45,126 +39,145 @@ To create a temporary `config.php` file:
...
+For an upgrade, you delete the `config.php` file. Once this file is added to your code, you should **not** delete it. If you need to remove or edit settings, you must edit the file manually.
 
-## Verify and upgrade your extensions {#extensions}
+### Upgrading extensions
 
-{%
-include note.html
-type='info'
-content='Review your third-party extension and module pages in Marketplace or other company sites to verify support for Magento Commerce and Magento Commerce Cloud version 2.2.
+Review your third-party extension and module pages in Marketplace or other company sites to verify support for {{site.data.var.ee}} and {{site.data.var.ece}}. If you need to upgrade any third-party extensions and modules that support, we recommend working in a new Integration branch with your extensions disabled.
```
Comment (meker12) [inline guides/v2.2/cloud/project/project-upgrade.md]:
```suggestion
Review your third-party extension and module pages in Marketplace or other company sites to verify support for {{site.data.var.ee}} and {{site.data.var.ece}}. If you need to upgrade any third-party extensions and modules, we recommend working in a new Integration branch with your extensions disabled.
```

File: guides/v2.2/cloud/project/project-upgrade.md [hunk 2]
```diff
@@ -45,126 +39,145 @@ To create a temporary `config.php` file:
...
 
 1. Wait for deployment to complete.
 
 1. Verify the upgrade in your Integration, Staging, or Production environment by using SSH to log in and check the version.
 
-    ```bash
-    php bin/magento --version
-    ```
+   ```bash
+   bin/magento --version
```
Comment (shiftedreality) [inline guides/v2.2/cloud/project/project-upgrade.md]: Is there any standard on running binary files? I see our hooks using PHP binary `php ...` and this line does not have it

Comment (hguthrie) [reply guides/v2.2/cloud/project/project-upgrade.md]: Yes, they removed PHP from a lot of the core documentation, but I haven't always allowed that to happen in the Cloud documentation. This one slipped by me. If you think it is best, I will revert to use PHP.

Comment (shiftedreality) [reply guides/v2.2/cloud/project/project-upgrade.md]: Just asking, don't see a big issue here

**Comment by hguthrie** [comment]
running tests

**Comment by hguthrie** [comment]
running tests

**Comment by hguthrie** [comment]
running tests

**Comment by hguthrie** [comment]
Holy smokes! It merged! Finally...

**Pull request merged**